File: Fortran/SaveSteadyStateOutput.f90
SUBROUTINE SaveSteadyStateOutput

USE Parameters
USE Globals
USE Procedures

IMPLICIT NONE
INTEGER		:: iy
CHARACTER	:: lstring*80


IF (Display>=1) write(*,*) 'Saving output to disk'

!grids
OPEN(3, FILE = trim(OutputDir) // 'agrid.txt', STATUS = 'replace'); CALL WriteMatrixLong(3,ngpa,1,agrid)
OPEN(3, FILE = trim(OutputDir) // 'bgrid.txt', STATUS = 'replace'); CALL WriteMatrixLong(3,ngpb,1,bgrid)
OPEN(3, FILE = trim(OutputDir) // 'ygrid.txt', STATUS = 'replace'); CALL WriteMatrixLong(3,ngpy,1,ygrid)
OPEN(3, FILE = trim(OutputDir) // 'adelta.txt', STATUS = 'replace'); CALL WriteMatrixLong(3,ngpa,1,adelta)
OPEN(3, FILE = trim(OutputDir) // 'bdelta.txt', STATUS = 'replace'); CALL WriteMatrixLong(3,ngpb,1,bdelta)

!initial steady state summary stats
OPEN(3, FILE = trim(OutputDir) // 'InitialSteadyStateParameters.txt', STATUS = 'replace')
	write(3,*) 'gam ',gam
	write(3,*) 'rho ',rho
	write(3,*) 'deathrate ',deathrate
	write(3,*) 'kappa0_w ',kappa0_w
	write(3,*) 'kappa1_w ',kappa1_w
	write(3,*) 'kappa2_w ',kappa2_w
	write(3,*) 'kappa3 ',kappa3
	write(3,*) 'kappa4_w ',kappa4_w
	write(3,*) 'kappa0_d ',kappa0_d
	write(3,*) 'kappa1_d ',kappa1_d
	write(3,*) 'kappa2_d ',kappa2_d
	write(3,*) 'kappa4_d ',kappa4_d
	write(3,*) 'corptax ',corptax

	write(3,*) 'ra ',equmINITSS%ra
	write(3,*) 'rb ',equmINITSS%rb
	write(3,*) 'rborr ',equmINITSS%rborr
	write(3,*) 'rcapital ',equmINITSS%rcapital
	write(3,*) 'wage ',equmINITSS%wage
	write(3,*) 'netwage ',equmINITSS%netwage
	write(3,*) 'bond ',equmINITSS%bond
	write(3,*) 'capital ',equmINITSS%capital
	write(3,*) 'equity ',equmINITSS%equity
	write(3,*) 'labor ',equmINITSS%labor
	write(3,*) 'output ',equmINITSS%output
	write(3,*) 'investment ',equmINITSS%investment
	write(3,*) 'govexp ',equmINITSS%govexp
	write(3,*) 'lumptransfer ',equmINITSS%lumptransfer
	write(3,*) 'labtax ',equmINITSS%labtax
	write(3,*) 'taxrev ',equmINITSS%taxrev
	write(3,*) 'govbond ',equmINITSS%govbond
	write(3,*) 'worldbond ',equmINITSS%worldbond
	write(3,*) 'fundbond ',equmINITSS%fundbond
	write(3,*) 'KYratio ',equmINITSS%KYratio
	write(3,*) 'KNratio ',equmINITSS%KNratio
	write(3,*) 'mc ',equmINITSS%mc
	write(3,*) 'rb ',equmINITSS%rb
	write(3,*) 'tfp ',equmINITSS%tfp
	write(3,*) 'pi ',equmINITSS%pi
	write(3,*) 'rnom ',equmINITSS%rnom
	write(3,*) 'priceadjust ',equmINITSS%priceadjust
	write(3,*) 'profit ',equmINITSS%profit
	write(3,*) 'dividend ',equmINITSS%dividend
	write(3,*) 'divrate ',equmINITSS%divrate

	write(3,*) 'borrwedge ',equmINITSS%borrwedge
	write(3,*) 'rho ',equmINITSS%rho
	write(3,*) 'fundlev ',equmINITSS%fundlev
	write(3,*) 'deprec ',deprec
	
	write(3,*) 'Ea ',statsINITSS%Ea
	write(3,*) 'Eb ',statsINITSS%Eb
	write(3,*) 'Ec ',statsINITSS%Ec
	write(3,*) 'Ehours ',statsINITSS%Ehours
	write(3,*) 'Elabor ',statsINITSS%Elabor
	write(3,*) 'Ed ',statsINITSS%Ed
	write(3,*) 'Ewage ',statsINITSS%Ewage
	write(3,*) 'Enetlabinc ',statsINITSS%Enetlabinc
	write(3,*) 'Egrosslabinc ',statsINITSS%Egrosslabinc
	write(3,*) 'Enetprofinc ',statsINITSS%Enetprofinc
	write(3,*) 'Egrossprofinc ',statsINITSS%Egrossprofinc
	write(3,*) 'Einc ',statsINITSS%Einc
	write(3,*) 'Enw ',statsINITSS%Enw
	write(3,*) 'FRACa0 ',statsINITSS%FRACa0
	write(3,*) 'FRACb0 ',statsINITSS%FRACb0
	write(3,*) 'FRACb0a0 ',statsINITSS%FRACb0a0
	write(3,*) 'FRACnw0 ',statsINITSS%FRACnw0
	write(3,*) 'FRACb0aP ',statsINITSS%FRACb0aP
	write(3,*) 'FRACa0close ',statsINITSS%FRACa0close
	write(3,*) 'FRACb0close ',statsINITSS%FRACb0close
	write(3,*) 'FRACb0a0close ',statsINITSS%FRACb0a0close
	write(3,*) 'FRACnw0close ',statsINITSS%FRACnw0close
	write(3,*) 'FRACbN ',statsINITSS%FRACbN
	write(3,*) 'EbN ',statsINITSS%EbN
	write(3,*) 'EbP ',statsINITSS%EbP
	write(3,*) 'Eadjcost ',statsINITSS%Eadjcost
	write(3,*) 'GINIa ',statsINITSS%GINIa
	write(3,*) 'GINIb ',statsINITSS%GINIb
	write(3,*) 'GINInw ',statsINITSS%GINInw
	write(3,*) 'GINIc ',statsINITSS%GINIc
	write(3,*) 'GINIinc ',statsINITSS%GINIinc
	write(3,*) 'Ec_bN ',statsINITSS%Ec_bN
	write(3,*) 'Ec_b0close ',statsINITSS%Ec_b0close
	write(3,*) 'Ec_b0far ',statsINITSS%Ec_b0far
	


CLOSE(3)

!initial steady state distributions and policy functions
CALL system ("mkdir -p " // trim(OutputDir) // "INITSS")

OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'PERCa.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,11,statsINITSS%PERCa)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'PERCb.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,11,statsINITSS%PERCb)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'PERCnw.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,11,statsINITSS%PERCnw)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'PERCc.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,11,statsINITSS%PERCc)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'PERCinc.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,11,statsINITSS%PERCinc)

OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'Ea_nwQ.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,4,statsINITSS%Ea_nwQ)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'Eb_nwQ.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,4,statsINITSS%Eb_nwQ)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'Ec_nwQ.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,4,statsINITSS%Ec_nwQ)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'Einc_nwQ.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,4,statsINITSS%Einc_nwQ)

OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'Ea_incQ.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,4,statsINITSS%Ea_incQ)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'Eb_incQ.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,4,statsINITSS%Eb_incQ)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'Ec_incQ.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,4,statsINITSS%Ec_incQ)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'Einc_incQ.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,1,4,statsINITSS%Einc_incQ)

DO iy = 1,ngpy
	IF (iy<10) WRITE(UNIT=lstring, FMT='(I1)') iy
	IF (iy >= 10) WRITE(UNIT=lstring, FMT='(I2)') iy

	OPEN(3, FILE = trim(OutputDir) // "INITSS/" // 'V_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%V(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'dep_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%d(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'con_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%c(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'sav_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%s(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'hour_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%h(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'bdot_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%bdot(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'gjoint_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%gjoint(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/" // 'B_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%B(iy)%val)

	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'ccum1_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,cumINITSS%ccum1(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'ccum2_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,cumINITSS%ccum2(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'ccum4_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,cumINITSS%ccum4(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'dcum1_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,cumINITSS%dcum1(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'dcum2_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,cumINITSS%dcum2(:,:,iy))
	OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'dcum4_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,cumINITSS%dcum4(:,:,iy))

	IF(ComputeDiscountedMPC==1) THEN
		OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'mpc_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%mpc(:,:,iy))
		OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'subeff1ass_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%subeff1ass(:,:,iy))
		OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'subeff2ass_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%subeff2ass(:,:,iy))
		OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'wealtheff1ass_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%wealtheff1ass(:,:,iy))
		OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'wealtheff2ass_INITSS_y' // trim(lstring) //'.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,solnINITSS%wealtheff2ass(:,:,iy))
	END IF	

END DO

OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'gamarg_INITSS.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpy,solnINITSS%gamarg)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'gbmarg_INITSS.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpb,ngpy,solnINITSS%gbmarg)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'gabmarg_INITSS.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,gabmarg)
OPEN(3, FILE = trim(OutputDir) // "INITSS/"  // 'gabcum_INITSS.txt', STATUS = 'replace'); CALL WriteMatrixExpon(3,ngpa,ngpb,gabcum)


END SUBROUTINE SaveSteadyStateOutput

File: mycode/SaveSteadyStateOutput.m
if Display; fprintf('Saving output to disk\n'); end

% grids
save([OutputDir,'grids'],'agrid','bgrid','ygrid','ydist','adelta','bdelta')

% initial steady state summary stats
fid = fopen(fullfile(OutputDir,'InitialSteadyStateParameters.txt'),'w');
fprintf(fid,'gam %.15g\n',gam);
fprintf(fid,'rho %.15g\n',rho);
fprintf(fid,'deathrate %.15g\n',deathrate);
fprintf(fid,'kappa0_w %.15g\n',kappa0_w);
fprintf(fid,'kappa1_w %.15g\n',kappa1_w);
fprintf(fid,'kappa2_w %.15g\n',kappa2_w);
fprintf(fid,'kappa3 %.15g\n',kappa3);
fprintf(fid,'kappa4_w %.15g\n',kappa4_w);
fprintf(fid,'kappa0_d %.15g\n',kappa0_d);
fprintf(fid,'kappa1_d %.15g\n',kappa1_d);
fprintf(fid,'kappa2_d %.15g\n',kappa2_d);
fprintf(fid,'kappa4_d %.15g\n',kappa4_d);
fprintf(fid,'corptax %.15g\n',corptax);

fprintf(fid,'ra %.15g\n',equmINITSS.ra);
fprintf(fid,'rb %.15g\n',equmINITSS.rb);
fprintf(fid,'rborr %.15g\n',equmINITSS.rborr);
fprintf(fid,'rcapital %.15g\n',equmINITSS.rcapital);
fprintf(fid,'wage %.15g\n',equmINITSS.wage);
fprintf(fid,'netwage %.15g\n',equmINITSS.netwage);
fprintf(fid,'bond %.15g\n',equmINITSS.bond);
fprintf(fid,'capital %.15g\n',equmINITSS.capital);
fprintf(fid,'equity %.15g\n',equmINITSS.equity);
fprintf(fid,'labor %.15g\n',equmINITSS.labor);
fprintf(fid,'output %.15g\n',equmINITSS.output);
fprintf(fid,'investment %.15g\n',equmINITSS.investment);
fprintf(fid,'govexp %.15g\n',equmINITSS.govexp);
fprintf(fid,'lumptransfer %.15g\n',equmINITSS.lumptransfer);
fprintf(fid,'labtax %.15g\n',equmINITSS.labtax);
fprintf(fid,'taxrev %.15g\n',equmINITSS.taxrev);
fprintf(fid,'govbond %.15g\n',equmINITSS.govbond);
fprintf(fid,'worldbond %.15g\n',equmINITSS.worldbond);
fprintf(fid,'fundbond %.15g\n',equmINITSS.fundbond);
fprintf(fid,'KYratio %.15g\n',equmINITSS.KYratio);
fprintf(fid,'KNratio %.15g\n',equmINITSS.KNratio);
fprintf(fid,'mc %.15g\n',equmINITSS.mc);
fprintf(fid,'rb %.15g\n',equmINITSS.rb);
fprintf(fid,'tfp %.15g\n',equmINITSS.tfp);
fprintf(fid,'pi %.15g\n',equmINITSS.pi);
fprintf(fid,'rnom %.15g\n',equmINITSS.rnom);
fprintf(fid,'priceadjust %.15g\n',equmINITSS.priceadjust);
fprintf(fid,'profit %.15g\n',equmINITSS.profit);
fprintf(fid,'dividend %.15g\n',equmINITSS.dividend);
fprintf(fid,'divrate %.15g\n',equmINITSS.divrate);

fprintf(fid,'borrwedge %.15g\n',equmINITSS.borrwedge);
fprintf(fid,'rho %.15g\n',equmINITSS.rho);
fprintf(fid,'fundlev %.15g\n',equmINITSS.fundlev);
fprintf(fid,'deprec %.15g\n',deprec);
	
fprintf(fid,'Ea %.15g\n',statsINITSS.Ea);
fprintf(fid,'Eb %.15g\n',statsINITSS.Eb);
fprintf(fid,'Ec %.15g\n',statsINITSS.Ec);
fprintf(fid,'Ehours %.15g\n',statsINITSS.Ehours);
fprintf(fid,'Elabor %.15g\n',statsINITSS.Elabor);
fprintf(fid,'Ed %.15g\n',statsINITSS.Ed);
fprintf(fid,'Ewage %.15g\n',statsINITSS.Ewage);
fprintf(fid,'Enetlabinc %.15g\n',statsINITSS.Enetlabinc);
fprintf(fid,'Egrosslabinc %.15g\n',statsINITSS.Egrosslabinc);
fprintf(fid,'Enetprofinc %.15g\n',statsINITSS.Enetprofinc);
fprintf(fid,'Egrossprofinc %.15g\n',statsINITSS.Egrossprofinc);
fprintf(fid,'Einc %.15g\n',statsINITSS.Einc);
fprintf(fid,'Enw %.15g\n',statsINITSS.Enw);
fprintf(fid,'FRACa0 %.15g\n',statsINITSS.FRACa0);
fprintf(fid,'FRACb0 %.15g\n',statsINITSS.FRACb0);
fprintf(fid,'FRACb0a0 %.15g\n',statsINITSS.FRACb0a0);
fprintf(fid,'FRACnw0 %.15g\n',statsINITSS.FRACnw0);
fprintf(fid,'FRACb0aP %.15g\n',statsINITSS.FRACb0aP);
fprintf(fid,'FRACa0close %.15g\n',statsINITSS.FRACa0close);
fprintf(fid,'FRACb0close %.15g\n',statsINITSS.FRACb0close);
fprintf(fid,'FRACb0a0close %.15g\n',statsINITSS.FRACb0a0close);
fprintf(fid,'FRACnw0close %.15g\n',statsINITSS.FRACnw0close);
fprintf(fid,'FRACbN %.15g\n',statsINITSS.FRACbN);
fprintf(fid,'EbN %.15g\n',statsINITSS.EbN);
fprintf(fid,'EbP %.15g\n',statsINITSS.EbP);
fprintf(fid,'Eadjcost %.15g\n',statsINITSS.Eadjcost);
fprintf(fid,'GINIa %.15g\n',statsINITSS.GINIa);
fprintf(fid,'GINIb %.15g\n',statsINITSS.GINIb);
fprintf(fid,'GINInw %.15g\n',statsINITSS.GINInw);
fprintf(fid,'GINIc %.15g\n',statsINITSS.GINIc);
fprintf(fid,'GINIinc %.15g\n',statsINITSS.GINIinc);
fprintf(fid,'Ec_bN %.15g\n',statsINITSS.Ec_bN);
fprintf(fid,'Ec_b0close %.15g\n',statsINITSS.Ec_b0close);
fprintf(fid,'Ec_b0far %.15g\n',statsINITSS.Ec_b0far);
	
fclose(fid);

% initial steady state distributions and policy functions
save([OutputDir,'solnINITSS'],'-struct','solnINITSS')
save([OutputDir,'cumINITSS'],'-struct','cumINITSS')
save([OutputDir,'equmINITSS'],'-struct','equmINITSS')
save([OutputDir,'statsINITSS'],'-struct','statsINITSS')